### PR TITLE
Homework 1.  Setting up the environment

### PR DESCRIPTION
**Pull request by kateika**
For this task you will need to setup a simple NodeJS server to implement server-side rendering. We recommend using the Express.

Please install Webpack and setup Babel for your application. Thy to make your configuration simple, as for the small SPA.

You will also need to install React, Redux, React-Redux, React-Router, Jest and Redux Form

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.DS_Store
+node_modules
+build
+production-build
```
Comment (yankouskia) [inline .gitignore]: + npm-debug.log

File: app.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const express = require('express');
```
Comment (yankouskia) [inline app.js]: as you use babel, you can connect it and use `es6 modules`

File: package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  "dependencies": {
+    "express": "^4.15.4",
+    "babel-loader": "^7.1.2",
+    "babel-core": "^6.26.0",
+    "babel-preset-es2015": "^6.24.1",
+    "babel-preset-react": "^6.24.1",
+    "extract-text-webpack-plugin": "^3.0.0",
+    "file-loader": "^0.11.2",
+    "html-webpack-plugin": "^2.30.1",    
+    "css-loader": "^0.28.7",
```
Comment (yankouskia) [inline package.json]: dev dep

File: package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "prod": "webpack -p --config ./webpack.production.config.js"
+  },
+  "keywords": [
+    "roulette"
+  ],
+  "author": "Kate She",
+  "license": "ISC",
+  "dependencies": {
+    "express": "^4.15.4",
+    "babel-loader": "^7.1.2",
```
Comment (yankouskia) [inline package.json]: dev dep

File: package.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    "babel-core": "^6.26.0",
```
Comment (yankouskia) [inline package.json]: remove

File: package.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    "babel-preset-es2015": "^6.24.1",
```
Comment (yankouskia) [inline package.json]: dev dep everything regarding build

File: package.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    "jest": "^21.0.2",
+    "react": "^15.6.1",
+    "react-dom": "^15.6.1",
+    "redux": "^3.7.2",
+    "style-loader": "^0.18.2",    
+    "webpack": "^3.5.6"
+  },
+  "devDependencies": {
+    "react-hot-loader": "^3.0.0-beta.7",
+    "react-redux": "^5.0.6",
```
Comment (yankouskia) [inline package.json]: to main deps everything regarding app's work

File: src/App.jsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import logo from './images/logo.png';
+import * as style from './styles/style.css';
+
+export default class App extends React.Component {
+
+    render() {
```
Comment (yankouskia) [inline src/App.jsx]: add please `editorconfig`. In one place you have 4-identation tab, in other files 2-identation

File: webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+        }]
+    },
+
+    plugins: [
+        new webpack.HotModuleReplacementPlugin(),
+        new ExtractTextPlugin("[name].css"),
+        new HtmlWebpackPlugin({
+            title: 'Test',
+            hash: true,
+            template: './index.html'
```
Comment (yankouskia) [inline webpack.config.js]: you can remove template. `HtmlWebpackPlugin ` will create it itself with link to `js` and `css` bundle

File: webpack.production.config.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
+const path = require('path');
+const webpack = require('webpack');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const ExtractTextPlugin = require("extract-text-webpack-plugin");
```
Comment (yankouskia) [inline webpack.production.config.js]: you have too much copypaste, better to create 1 webpack config with dependence to env variable like `NODE_ENV`. For example:
```js
{
  // ...
  devtool: process.env.NODE_ENV === 'production' ? 'eval' : null,
  // ...
}
```

File: webpack.config.js [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        './index.jsx'
+    ],
+
+    output: {
+        filename: 'bundle.js',
+        path:  path.resolve(__dirname, 'build'),
+    },
+
+    resolve: {
+        extensions: ['.js', '.jsx']
```
Comment (yankouskia) [inline webpack.config.js]: you can use just `js` everywhere

File: webpack.config.js [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+    },
+
+    devServer: {
```
Comment (yankouskia) [inline webpack.config.js]: usually dev server configuration is moved to another folder/file. It's just webpack config

File: webpack.config.js [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+    },
+
+    module: {
+        rules: [{
+            test: /\.jsx?$/,
+            exclude: /node_modules/,
+            use: {
+                loader: 'babel-loader',
+                options: {
+                    plugins: ["babel-plugin-transform-react-jsx"]
```
Comment (yankouskia) [inline webpack.config.js]: dont' count plugins here, you have already `.babelrc`

File: .babelrc [hunk 14]
```diff
@@ -0,0 +1,14 @@
+{
+    "presets": [
+        [
+            "es2015", {
+                "modules": false
```
Comment (yankouskia) [inline .babelrc:5]: 👍

File: src/index.html [hunk 15]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
```
Comment (yankouskia) [inline src/index.html]: remove

File: .gitignore (hunk 1 above)
Comment (kateika) [reply .gitignore]: done

File: package.json (hunk 4 above)
Comment (kateika) [reply package.json]: done

File: package.json (hunk 5 above)
Comment (kateika) [reply package.json]: done

File: package.json (hunk 6 above)
Comment (kateika) [reply package.json]: done

File: package.json (hunk 3 above)
Comment (kateika) [reply package.json]: done

File: webpack.config.js (hunk 11 above)
Comment (kateika) [reply webpack.config.js]: done

**Pull request merged**